wrangler-core/src/main/java/io/cdap/wrangler/parser/GrammarBasedParser.java

```diff
@@ -95,8 +98,8 @@ public List<Directive> parse()
       return result;
     } catch (CompileException e) {
       throw new DirectiveParseException(e.getMessage(), e.iterator(), e);
-    } catch (DirectiveLoadException | DirectiveNotFoundException | DirectiveParseException e) {
-      throw e;
+    } catch (DirectiveLoadException | DirectiveNotFoundException e) {
+      throw new RecipeException(e.getMessage(), e, directiveIndex.get());
     } catch (Exception e) {
       throw new DirectiveParseException(e.getMessage(), e);
```

Comment:
In that case, do you want to add directive number to DirectiveParseException and pass it from here? It will make this function uniformly capture directive index for all kind of errors.